backend/tournesol/models/comparisons.py

```diff
@@ -159,4 +160,13 @@ def save(self, *args, **kwargs):
                 f"The absolute value of the score {self.score} given to the criterion "
                 f"{self.criteria} can't be greater than the value of score_max {self.score_max}."
             )
+
+    def clean(self):
+        try:
+            self._validate_score_max()
+        except (TypeError, ValueError) as err:
+            raise ValidationError({"score_max": err.args[0]}) from err
+
+    def save(self, *args, **kwargs):
+        self._validate_score_max()
         return super().save(*args, **kwargs)
```

Comment:
Maybe something we missed in https://github.com/tournesol-app/tournesol/pull/1947, it does not seem like the case when an invalid score/score_max is tested in "backend/tournesol/tests/test_api_comparison.py". As it is now, it may return a 500 response, I think.

Reply:
Indeed, I created an issue to fix this: https://github.com/tournesol-app/tournesol/issues/2001